components/brave_ads/core/internal/account/confirmations/confirmation_type.cc

```diff
@@ -28,101 +29,67 @@ constexpr char kConversionType[] = "conversion";
 
 }  // namespace
 
-ConfirmationType::ConfirmationType() = default;
-
-ConfirmationType::ConfirmationType(const std::string& value) {
+ConfirmationType ParseConfirmationType(std::string_view value) {
   if (value == kUndefinedType) {
-    value_ = kUndefined;
+    return ConfirmationType::kUndefined;
   } else if (value == kClickedType) {
-    value_ = kClicked;
+    return ConfirmationType::kClicked;
   } else if (value == kDismissedType) {
-    value_ = kDismissed;
+    return ConfirmationType::kDismissed;
   } else if (value == kViewedType) {
-    value_ = kViewed;
+    return ConfirmationType::kViewed;
   } else if (value == kServedType) {
-    value_ = kServed;
+    return ConfirmationType::kServed;
   } else if (value == kTransferredType) {
-    value_ = kTransferred;
+    return ConfirmationType::kTransferred;
   } else if (value == kSavedType) {
-    value_ = kSaved;
+    return ConfirmationType::kSaved;
   } else if (value == kFlaggedType) {
-    value_ = kFlagged;
+    return ConfirmationType::kFlagged;
   } else if (value == kUpvotedType) {
-    value_ = kUpvoted;
+    return ConfirmationType::kUpvoted;
   } else if (value == kDownvotedType) {
-    value_ = kDownvoted;
+    return ConfirmationType::kDownvoted;
   } else if (value == kConversionType) {
-    value_ = kConversion;
-  } else {
-    SCOPED_CRASH_KEY_STRING32("ConfirmationType", "value", value);
-    NOTREACHED() << "Unexpected value for ConfirmationType: " << value;
+    return ConfirmationType::kConversion;
   }
+  SCOPED_CRASH_KEY_STRING32("ConfirmationType", "value", value);
+  NOTREACHED() << "Unexpected value for ConfirmationType: " << value;
+  return ConfirmationType::kUndefined;
 }
 
-ConfirmationType::Value ConfirmationType::value() const {
-  return value_;
-}
-
-std::string ConfirmationType::ToString() const {
-  switch (value_) {
-    case kUndefined: {
+const char* ToString(ConfirmationType type) {
+  switch (type) {
+    case ConfirmationType::kUndefined:
```

Comment:
Would be great to keep the curly braces, however please see other comment about our ongoing efforts.